### PR TITLE
Feature/finalize error handling sa

### PR DESCRIPTION
**Pull request by StevieAmb**
What (if any) features are you implementing?
- Error handling feature for a status 500 and 400 error

What (if anything) did you refactor?
- Refactored how we were using error handling component

Were there any issues that arose?
- We wondered if we would be able to give a component two parents, and realized we could if it was only being affected by rendering one time

Is there anything that you need from your teammate?
- Keep being awesome! :D 

Any other comments, questions, or concerns?
- No

### CODE REVIEW

File: cypress/integration/error_handling_spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  it('User should see error message', () => {
+    cy.get('.error-message').should('be.visible')
+  })
+
+  it('User should see back to main page button, and be able to click it to return to main page', () => {
+    cy.get('.return-to-main').contains('Back To Main Page').click()
+    .url().should("eq", "http://localhost:3000/");
+  })
+
+})
```
Comment (mr-ryan12) [inline cypress/integration/error_handling_spec.js:26]: Testing looks great! We can look into creating tests for 400 and 500 errors at some point, but this is a great start!

**Review by tonycarpenter21 — APPROVED** [review]
This testing looks great!

**Review by Jayquintana — APPROVED** [review]
Good job stevie!!

File: src/404/404.tsx [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+}
+
+const ErrorHandling: React.FC<EProps> = ({message}) => {
   return (
     <section className="error">
-      <img src="https://media2.giphy.com/media/rYEAkYihZsyWs/giphy.gif?cid=ecf05e47mxw0p2ev880va40mrexkbc8mm0gvmprvp8elek15&rid=giphy.gif&ct=g" alt="Bob Ross 404 error"/>
-      <h1 className="error-message">Page not found. Let's just paint a happy little 404 message right here!</h1>
+      <img className='error-pic' src="https://media2.giphy.com/media/rYEAkYihZsyWs/giphy.gif?cid=ecf05e47mxw0p2ev880va40mrexkbc8mm0gvmprvp8elek15&rid=giphy.gif&ct=g" alt="Bob Ross 404 error"/>
+      <h1 className="error-message">{message}</h1>
+      <NavLink to="/" className='return-to-main'>Back To Main Page</NavLink>
```
Comment (mr-ryan12) [inline src/404/404.tsx:14]: Love the Bob Ross gif!

File: src/App.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ class App extends Component<{}, State> {
           <Switch>
             < Route exact path='/' render={() => <Categories categories={this.state.categories}/> }/>
             < Route exact path='/:category' render={({ match }) => this.checkCategory(match.params.category) } />
-            < Route render={() => <ErrorHandling /> } />
+            < Route render={() => <ErrorHandling message='Ya blew it!' /> } />
```
Comment (mr-ryan12) [inline src/App.tsx:39]: Maybe we can change the message to reflect the error 😂

**Pull request merged**